mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/dirtiness/building/impl/PojoImplicitReindexingResolverBuildingHelper.java

```diff
@@ -157,4 +158,27 @@ ReindexOnUpdate getReindexOnUpdate(ReindexOnUpdate parentReindexOnUpdate,
 			return reindexOnUpdateOptional.orElse( ReindexOnUpdate.DEFAULT );
 		}
 	}
+
+	Set<PojoModelPathValueNode> getDerivedFrom(PojoTypeModel<?> typeModel, String propertyName,
+			ContainerValueExtractorPath extractorPath) {
+		PojoTypeAdditionalMetadata typeAdditionalMetadata =
+				typeAdditionalMetadataProvider.get( typeModel.getRawType() );
+		Set<PojoModelPathValueNode> derivedFrom =
+				typeAdditionalMetadata.getPropertyAdditionalMetadata( propertyName )
+						.getValueAdditionalMetadata( extractorPath )
+						.getDerivedFrom();
+		if ( derivedFrom.isEmpty() ) {
+			if ( extractorBinder.isDefaultExtractorPath(
+					introspector,
+					typeModel.getProperty( propertyName ).getTypeModel(),
+					extractorPath
+			) ) {
```

Comment:
I see a lot of these things, wondering if it should be hidden somehow? I.e. here, I would have expected only one call to `getValueAdditionalMetadata()...` and I would have expected it to do the right thing for me.

Is there a special reason to not do this?

Reply:
> I could hide the isDefaultExtractorPath bit inside getValueAdditionalMetadata, but then I'd have to pass more arguments than seem reasonable to that method (the container value extractor binder plus the bound path value node, in particular).

Sorry, I should have been more clear, that's what I had in mind. I think it highly depends on if we always have to consider the default extractor path case. If so, I think it has some value to hide it so that nobody has to think about it: you just pass the required arguments and you get the right info.

It might not make sense if we could have to get the `getValueAdditionalMetadata` info without considering this case.

Reply:
I just had a look, and it turns out we can't really move the code to `getValueAdditionalMetadata` because, in order to decide wether we have to check the default path, we have to retrieve some value and determine whether it's empty or not. In some case the value is the inverse path, in some cases it's the "derivedFrom", ... It depends.
So unless you're ok with adding some more lambdas, I think we'll have to keep it that way :)